R/cat_add_data.R

```diff
@@ -0,0 +1,413 @@
+#' Add data
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' `elic_cat_add_data()` adds data to an [elic_cat] object from different
+#' sources.
+#'
+#' @param x an object of class [elic_cat].
+#' @param mechanism character string that indicates the machanism to which the
+#' data belongs.
+#' @inheritParams elic_cont_add_data
+#'
+#' @section Data format:
+#'
+#' For each mechanism, data are expected to have five columns, built as follows:
+#' * The first column of the data should hold the names of the experts. The name
+#' of each expert should be repeated as many times as many times as the number
+#' of impact levels and sites.
+#' (i.e. each expert should appear \eqn{number\ of\ levels
+#' \cdot number\ of\ sites} times).
+#' * The second column should be the names of the levels of impact considered in
+#' the elicitation. Each block of levels of impact should be repeated as many
+#' times as the number of sites investigated/considered.
+#' * The third column should hold the names of the sites considered in the
+#' study. The name of each site should be repeated as many times as the number
+#' of levels of impact considered.
+#' * The fourth column should be the experts confidence in their own estimates
+#' (given in percent). Experts should estimate how confident they are in their
+#' estimates for each block of levels of impact and for each site. Therefore,
+#' expert confidence estimates should be repeated as many times as the number of
+#' levels of impact considered.
+#' * The final column should be the estimates of each expert for each site and
+#' impact level.
+#'
+#' The name of the columns is not important, `elic_cat_add_data()` will
+#' overwrite them according to the following convention:
+#' * The first column will be renamed `name`, the second column `level`, the
+#' third column `site`, the fourth column `confidence`, and the fifth column
+#' `estimate`.
+#'
+#' Here is an example of how data correctly formatted should look like, for an
+#' elicitation with two experts, two levels of impact,two sites,  and one
+#' mechanism:
```

Comment:
Lines 42-44: I would write it like this:

> Here is an example of data correctly formatted for an elicitation with two levels of impact and two sites (only one expert is shown):

That is for one mechanism should be clear, since you can add data only to one mechanism.